sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/options/EncryptOptions.java

```diff
@@ -0,0 +1,130 @@
+     */
+    byte[] additionalAuthenticatedData;
+
+    /**
+     * Factory method to create an instance of {@link EncryptOptions} with the given parameters.
+     *
+     * @param algorithm The algorithm to be used for encryption.
+     * @param plaintext The content to be encryption.
+     * @return The {@link EncryptOptions}.
+     */
+    public static EncryptOptions createOptions(EncryptionAlgorithm algorithm, byte[] plaintext) {
+        return new EncryptOptions(algorithm, plaintext);
+    }
+
+    /**
+     * Factory method to create an instance of {@link AesCbcEncryptOptions} with the given parameters.
+     *
+     * @param algorithm The algorithm to be used for encryption.
+     * @param plaintext The content to be encryption.
+     * @return The {@link AesCbcEncryptOptions}.
+     */
+    public static AesCbcEncryptOptions createAesCbcOptions(EncryptionAlgorithm algorithm, byte[] plaintext) {
+        return new AesCbcEncryptOptions(algorithm, plaintext);
+    }
+
+    /**
+     * Factory method to create an instance of {@link AesGcmEncryptOptions} with the given parameters.
+     *
+     * @param algorithm The algorithm to be used for encryption.
+     * @param plaintext The content to be encryption.
+     * @param iv Initialization vector for the encryption operation.
+     * @return The {@link AesGcmEncryptOptions}.
+     */
+    public static AesGcmEncryptOptions createAesGcmOptions(EncryptionAlgorithm algorithm, byte[] plaintext, byte[] iv) {
+        return new AesGcmEncryptOptions(algorithm, plaintext, iv);
+    }
+
+    /**
+     * Creates an instance of {@link EncryptOptions} with the given parameters.
+     *
+     * @param algorithm The algorithm to be used for encryption.
+     * @param plainText The content to be encrypted.
+     */
+    EncryptOptions(EncryptionAlgorithm algorithm, byte[] plainText) {
+        Objects.requireNonNull(algorithm, "'algorithm cannot be null'");
+        Objects.requireNonNull(plainText, "'plaintext' cannot be null");
+
+        this.algorithm = algorithm;
+        this.plainText = new byte[plainText.length];
+        System.arraycopy(plainText, 0, this.plainText, 0, plainText.length);
+    }
+
+    /**
+     * The algorithm to be used for encryption.
+     *
+     * @return The algorithm to be used for encryption.
+     */
+    public EncryptionAlgorithm getAlgorithm() {
+        return algorithm;
+    }
+
+    /**
+     * Get the content to be encrypted.
+     *
+     * @return The content to be encrypted.
+     */
+    public byte[] getPlainText() {
+        if (plainText == null) {
+            return null;
+        } else {
+            return plainText.clone();
+        }
+    }
+
+    /**
+     * Get the initialization vector to be used in the encryption operation using a symmetric algorithm.
+     *
+     * @return The initialization vector.
+     */
+    public byte[] getIv() {
+        if (iv == null) {
+            return null;
+        } else {
+            return iv.clone();
```

Comment:
Clone again? Also, this needs to be generated at the proper size with a proper RNG if null.

Reply:
It gets generated if not provided [here](https://github.com/vcolin7/azure-sdk-for-java/blob/ff77ba13802b18d690436a3ec2534f732fb3bc47/sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/SymmetricKeyCryptographyClient.java#L85).